core/src/main/java/org/jsmart/zerocode/core/engine/mocker/RestEndPointMocker.java

```diff
@@ -95,30 +97,38 @@ public static void stopWireMockServer() {
     }
 
     private static MappingBuilder createDeleteRequestBuilder(MockStep mockStep) {
-        final MappingBuilder requestBuilder = delete(urlEqualTo(mockStep.getUrl()));
+        final MappingBuilder requestBuilder = delete(buildUrlPattern(mockStep.getUrl()));
         return createRequestBuilderWithHeaders(mockStep, requestBuilder);
     }
 
     private static MappingBuilder createPatchRequestBuilder(MockStep mockStep) {
-        final MappingBuilder requestBuilder = patch(urlEqualTo(mockStep.getUrl()));
+        final MappingBuilder requestBuilder = patch(buildUrlPattern(mockStep.getUrl()));
         return createRequestBuilderWithHeaders(mockStep, requestBuilder);
     }
 
     private static MappingBuilder createPutRequestBuilder(MockStep mockStep) {
-        final MappingBuilder requestBuilder = put(urlEqualTo(mockStep.getUrl()));
+        final MappingBuilder requestBuilder = put(buildUrlPattern(mockStep.getUrl()));
         return createRequestBuilderWithHeaders(mockStep, requestBuilder);
     }
 
     private static MappingBuilder createPostRequestBuilder(MockStep mockStep) {
-        final MappingBuilder requestBuilder = post(urlEqualTo(mockStep.getUrl()));
+        final MappingBuilder requestBuilder = post(buildUrlPattern(mockStep.getUrl()));
         return createRequestBuilderWithHeaders(mockStep, requestBuilder);
     }
 
     private static MappingBuilder createGetRequestBuilder(MockStep mockStep) {
-        final MappingBuilder requestBuilder = get(urlEqualTo(mockStep.getUrl()));
+        final MappingBuilder requestBuilder = get(buildUrlPattern(mockStep.getUrl()));
         return createRequestBuilderWithHeaders(mockStep, requestBuilder);
     }
 
+    private static UrlPattern buildUrlPattern(String url) {
```

Comment:
Sorry I could not find tests related to this, I mean not for this private function but for the change in logic for url matching.  
Can you please point to the tests/scenarios where this strict matching vs path matching is needed

Reply:
@MrXavier  please add appropriate tests for this one.